### PR TITLE
RSDK-5024: Add Sensor interface in resource.go that implements getReadings and embed that interface into powersensor movement sensor and sensor

### PR DESCRIPTION
**Pull request by martha-johnston**
Added `getReadings` to client/server files and moved the sensor interface to resource.go. I was a little confused by this ticket and I'm not confident I did this correctly so very open to suggestions and changes.

I believe this should also cover RSDK-4792 and RSDK-4344

### CODE REVIEW

File: testutils/inject/powersensor.go [hunk 1]
```diff
@@ -10,11 +10,12 @@ import (
...
-	VoltageFunc func(ctx context.Context, extra map[string]interface{}) (float64, bool, error)
-	CurrentFunc func(ctx context.Context, extra map[string]interface{}) (float64, bool, error)
-	PowerFunc   func(ctx context.Context, extra map[string]interface{}) (float64, error)
-	DoFunc      func(ctx context.Context, cmd map[string]interface{}) (map[string]interface{}, error)
+	name         resource.Name
+	VoltageFunc  func(ctx context.Context, extra map[string]interface{}) (float64, bool, error)
+	CurrentFunc  func(ctx context.Context, extra map[string]interface{}) (float64, bool, error)
+	PowerFunc    func(ctx context.Context, extra map[string]interface{}) (float64, error)
+	DoFunc       func(ctx context.Context, cmd map[string]interface{}) (map[string]interface{}, error)
+	ReadingsFunc func(ctx context.Context, extra map[string]interface{}) (map[string]interface{}, error)
```
Comment (randhid) [inline testutils/inject/powersensor.go:18]: optional - keep DoFunc last in all injects.

**Review by randhid — APPROVED** [review]
LGTM! Few nits here and there, feel free to address whenever.

File: components/sensor/sensor.go [hunk 2]
```diff
@@ -39,8 +37,9 @@ func Named(name string) resource.Name {
 // of some thing that it is sensing.
 type Sensor interface {
 	resource.Resource
+	resource.Sensor
 	// Readings return data specific to the type of sensor and can be of any type.
-	Readings(ctx context.Context, extra map[string]interface{}) (map[string]interface{}, error)
+	// Readings(ctx context.Context, extra map[string]interface{}) (map[string]interface{}, error)
```
Comment (susmitaSanyal) [inline components/sensor/sensor.go:42]: why not remove line 42?

**Review by susmitaSanyal — APPROVED** [review]
LGTM!

**Pull request merged**